File: concrete/src/main/java/pcb/uwu/unit/derived/optics/LuxUnit.java
package pcb.uwu.unit.derived.optics;

import static pcb.uwu.unit.base.CandelaUnit.CANDELA;
import static pcb.uwu.unit.base.MeterUnit.METER;
import static pcb.uwu.unit.scalar.SteradianUnit.STERADIAN;

public class LuxUnit extends IlluminanceUnit {

	public static final LuxUnit LUX = new LuxUnit();

	public LuxUnit() {
		super(CANDELA, STERADIAN, METER);
	}

	@Override
	public String getSymbol() {
		return "lx";
	}

	@Override
	public String getSingularName() {
		return "lux";
	}

	@Override
	public String getPluralName() {
		return "lux";
	}
}

File: concrete/src/main/java/pcb/uwu/unit/derived/optics/LuxUnit.kt
package pcb.uwu.unit.derived.optics

import pcb.uwu.unit.base.CandelaUnit
import pcb.uwu.unit.base.MeterUnit
import pcb.uwu.unit.scalar.SteradianUnit

object LuxUnit
    : IlluminanceUnit(CandelaUnit, SteradianUnit, MeterUnit)
{
    override val symbol = "lx"
    override val singularName = "lux"
    override val pluralName = "lux"
}